server/src/main/java/org/opensearch/index/store/remote/filecache/FileCache.java

```diff
@@ -123,4 +133,35 @@ public CacheUsage usage() {
     public CacheStats stats() {
         return theCache.stats();
     }
+
+    /**
+     * Restores the file cache instance performing a folder scan of the
+     * {@link org.opensearch.index.store.remote.directory.RemoteSnapshotDirectoryFactory#LOCAL_STORE_LOCATION}
+     * directory within the provided file cache path.
+     * @param fileCacheNodePath path pointing to the file cache directory
+     * @param fileCache the file cache instance to be restored
+     */
+    public static void restoreFileCache(final NodeEnvironment.NodePath fileCacheNodePath, final FileCache fileCache) {
```

Comment:
Related to another comment, but could you rewrite this as:

```
    public void restoreFileCache(List<Path> cacheDataPaths) {
        cacheDataPaths.stream()
            .filter(Files::isDirectory)
            .map(path -> path.resolve(LOCAL_STORE_LOCATION))
            .filter(Files::isDirectory)
            .flatMap(dir -> {
                try {
                    return Files.list(dir);
                } catch (IOException e) {
                    throw new OpenSearchException("File cache corrupted. Please clear the file cache for node startup.", e);
                }
            })
            .filter(Files::isRegularFile)
            .forEach(path -> put(path.toAbsolutePath(), new FileCachedIndexInput.ClosedIndexInput(Files.size(cachedFilePath))));
    }
```

Reply:
I followed the non-stream approach for streamlined error handling. This would be equivalent to the following with streams - 

```
public void restoreFileCache(List<Path> cacheDataPaths) {
        cacheDataPaths.stream()
            .filter(Files::isDirectory)
            .map(path -> path.resolve(LOCAL_STORE_LOCATION))
            .filter(Files::isDirectory)
            .flatMap(dir -> {
                try {
                    return Files.list(dir);
                } catch (IOException e) {
                    throw new OpenSearchException("File cache corrupted. Please clear the file cache for node startup.", e);
                }
            })
            .filter(Files::isRegularFile)
            .forEach(path -> {
                try {
                    put(path.toAbsolutePath(), new FileCachedIndexInput.ClosedIndexInput(Files.size(path)));
                } catch (IOException e) {
                    throw new OpenSearchException("File cache corrupted. Please clear the file cache for node startup.", e);
                }
            });
    }
```

Reply:
You don't need the duplicate try-catch if you use `toAbsolutePath()` in place of `toRealPath()`

Reply:
It's the `Files.size` that throws the IOException.